dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/MemoryAllocation.java

```diff
@@ -38,12 +38,14 @@ public void run(Program program) {
     }
 
     private void processMallocs(Program program) {
-        for (Malloc malloc : program.getEvents(Malloc.class)) {
-            final MemoryObject allocatedObject = program.getMemory().allocate(getSize(malloc), false);
-            final Local local = EventFactory.newLocal(malloc.getResultRegister(), allocatedObject);
-            local.addTags(Tag.Std.MALLOC);
-            local.copyAllMetadataFrom(malloc);
-            malloc.replaceBy(local);
+        for (Thread thread : program.getThreads()) {
+            for (Malloc malloc : thread.getEvents(Malloc.class)) {
```

Comment:
can you use `program.getThreadEvents`?

Reply:
Actually, this code used to call `program.getThreadEvents` but I changed it in an attempt to get rid of that function. It turned out to be quite difficult to replace, so I might as well revert my changes on that one.